tests/conversion_test.py

```diff
@@ -546,6 +547,32 @@ def test_pytket_qir_20(profile: QIRProfile) -> None:
     run_qir_gen_and_check(circ, "test_pytket_qir_20", profile=profile)
 
 
+@pytest.mark.parametrize(
+    "profile",
+    [
+        QIRProfile.ADAPTIVE,
+        QIRProfile.AZUREADAPTIVE,
+        QIRProfile.PYTKET,
+        QIRProfile.ADAPTIVE_CREGSIZE,
+    ],
+)
+def test_pytket_qir_21(profile: QIRProfile) -> None:
```

Comment:
This testcase is not really ideal because most of the classical operation disappear after the classical optimisation. Could you maybe add one initial quantum measurement to one of the registers so we see the classical ops in the QIR?

Reply:
Yes, will do.

Reply:
Done. Are the QIR files as you would expect?

Reply:
Looks good!